server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1237,7 +1237,7 @@ public boolean isSegRepLocalEnabled() {
     }
 
     public boolean isSegRepWithRemoteEnabled() {
-        return isSegRepEnabledOrRemoteNode() && isRemoteStoreEnabled();
+        return isSegRepEnabledOrRemoteNode() && isRemoteNode();
```

Comment:
Lets rephrase it -> `ReplicationType.SEGMENT.equals(replicationType) && isRemoteNode()`

Reply:
Done in latest Rev